src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int BASE_SIZE = 10;
+    private static final String INDEX_EXCEPTION_MESSAGE = "Provided index "
+            + "is < 0 or > list size. Please, provide valid index.";
+    private Object[] objects;
+    private int size;
+
+    public ArrayList() {
+        this.objects = new Object[BASE_SIZE];
+    }
+
+    private void checkFreeSpace(int neededSpace) {
+        if (neededSpace > objects.length) {
+            int newLength = (int)(objects.length * 1.5);
+            if (newLength < neededSpace) {
+                newLength = neededSpace;
+            }
```

Comment:
Remove double if construction, just check current capacity and if you need to grow your array use your arraycopy with objectLength * 1.5 argument

Reply:
and if you are comparing your array to the list and array.length + list.size > array.length * 1.5?